### DISCUSSION TITLE
C: Get assignement to struct field

### DISCUSSION THREAD

**Discussion by 0xTen** [github/codeql#12359] [General]
Hey!

I'm working with C atm and I was trying to figure out a way of getting all variables that are assigned to a particular struct field:

```
struct a {
    void *b;
};

void some_func(struct a *c){
    [...]
    void *d = c->b;
    [...]
}
```

I would want my query to find that variable `d` has an assignment to the `b` field of an `a` struct.
First day using ql, still getting the hang of it

**Comment by 0xTen** [comment]
I think I solved it:

```
import cpp

from Struct struct, Field field, VariableAccess access
where
    struct.getSize() != 0 and
    struct.getName() = "a" and

	field = struct.getAField() and
    field.getName()= "b" and

    access = field.getAnAccess()
    

select access, "a->b accessed here"
```

**Reply by hmakholm** [reply]
That looks right. Glad you figured it out!

(It shouldn't be necessary to filter on `struct.getSize() != 0`, unless you somehow have a zero-length struct with the same name and a field (itself zero-length) with the same name that you need to avoid matching).
